Compilation error in RecastDemo: multiple BuildContext

multiple definition of `BuildContext::~BuildContext()':

Linking executable: ..\..\Bin\RecastDemo.exe
lib\Debug/libDetourTileCache.a(DetourTileCache.o):DetourTileCache.cpp:(.text$_ZN12BuildContextD1Ev[BuildContext::~BuildContext()]+0x0): multiple definition of `BuildContext::~BuildContext()'
obj\Debug\RecastDemo\Source\SampleInterfaces.o:RecastDemo/Source/SampleInterfaces.cpp:26: first defined here
using master branch.

Indeed BuildContext is declared twice;
- once in the lib DetourCacheTile
- once in SampleInterfaces.cpp/h

Example of solution: rename BuildContext struct in the DetourCacheTile lib by dtcDetourCacheTile or use a namespace.

@memononen 
I encountered the same issue.
The error shows the line number where the conflicting BuildContext  occurs in DetourTileCache.cpp.
````
Linking RecastDemo
lib/Debug/libDetourTileCache.a(DetourTileCache.o):c:\recastnavigation\RecastDemo\Build\gmake/../../../DetourTileCache/Source/DetourTileCache.cpp:46:
multiple definition of `BuildContext::~BuildContext()'
obj/Debug/RecastDemo/SampleInterfaces.o:c\recastnavigation\RecastDem
o\Build\gmake/../../Source/SampleInterfaces.cpp:26: first defined here
collect2.exe: error: ld returned 1 exit status
make[1]: *** [../../Bin/RecastDemo.exe] Error 1
make: *** [RecastDemo] Error 2
````
I'm using mingw32 on windows 7 to compile commit 1dd5cf1883d61e723fef3d4957cf758c50e7a52b which is the current head of the repo.

--- a/DetourTileCache/Source/DetourTileCache.cpp
+++ b/DetourTileCache/Source/DetourTileCache.cpp
@@ -40,10 +40,10 @@ inline int computeTileHash(int x, int y, const int mask)
 }
 
 
-struct BuildContext
+struct NavMeshTileBuildContext
 {
-	inline BuildContext(struct dtTileCacheAlloc* a) : layer(0), lcset(0), lmesh(0), alloc(a) {}
-	inline ~BuildContext() { purge(); }
+	inline NavMeshTileBuildContext(struct dtTileCacheAlloc* a) : layer(0), lcset(0), lmesh(0), alloc(a) {}
+	inline ~NavMeshTileBuildContext() { purge(); }
 	void purge()
 	{
 		dtFreeTileCacheLayer(alloc, layer);
@@ -587,7 +587,7 @@ dtStatus dtTileCache::buildNavMeshTile(const dtCompressedTileRef ref, dtNavMesh*
 	
 	m_talloc->reset();
 	
-	BuildContext bc(m_talloc);
+	NavMeshTileBuildContext bc(m_talloc);
 	const int walkableClimbVx = (int)(m_params.walkableClimb / m_params.ch);
 	dtStatus status;
 	

